server/aesdsocket.c

```diff
@@ -0,0 +1,242 @@
+        			
+        			exit (-1);
+        		}
+    			/*set the working directory to the root directory*/
+    			if(chdir("/") == -1)
+    			{
+    				perror("error changing directory");
+    				exit (-1);
+    			} 
+    			/*redirect fd's 0,1,2 to /dev/null*/
+    			open("/dev/null", O_RDWR); /*stdin*/
+    			dup(0);				/*stdout*/
+    			dup(0);
+    			
+    			printf("Daemon created\n");
+    			
+    		}
+            else 
+            {
+                exit (0);
+            }
+        }
+    }
+
+    
+    int listenfd;
+    listenfd = listen(sockfd, 5);
+    printf("Listening successful\n");
+
+    if (listenfd == -1)
+    {
+        perror("Error in Listening process \n");
+        return -1;
+    }
+
+    
+    /*file_fd = open("/var/tmp/aesdsocketdata",O_RDWR | O_CREAT | O_TRUNC, S_IRUSR | S_IWUSR | S_IRGRP |S_IWGRP | S_IROTH | S_IWOTH, 0777);*/
+    file_fd = open("/var/tmp/aesdsocketdata",O_RDWR | O_CREAT | O_TRUNC, 0777);
+    
+    if (file_fd == -1)
+    {
+        perror("Error in opening file descriptor.");
+    }
+    
+    int file_size = 0;
+    
+    char receive_buffer[BUF_SIZE];
+    
+    while(1)
+    {
+        client_len = sizeof(clientadd);
+        clientfd = accept(sockfd, (struct sockaddr *)&clientadd, &client_len);
+        if (clientfd == -1)
+        {
+            perror("error accepting\n");
+            return -1;
+        }
+        printf("Accepted connection from %s\n",inet_ntoa(clientadd.sin_addr));
+        syslog(LOG_DEBUG,"Accepted connection from %s\n",inet_ntoa(clientadd.sin_addr));  
+
+        bool status = false;
+        int buffer = BUF_SIZE;
+        int buffer_size = 0;
+        char *write_buf = malloc(sizeof(char) * buffer);
+
+        while(!status)
+        {
+            // receive the data from client
+            rec = recv(clientfd, receive_buffer, BUF_SIZE, 0);
+            if (rec == 0)
+            {
+                status = true;
+            } 
+            for (int i = 0; i < rec; i++)
+            {
+                
+                // appneding new line
+ 		if (receive_buffer[i] == '\n')
+                {
+                    status = true;
+                }
+            }
+            if ((buffer - buffer_size) < rec)
+            {
+                buffer = buffer +rec;
+                write_buf = (char *) realloc(write_buf, sizeof(char) * buffer);
+            }
+            memcpy(write_buf + buffer_size, receive_buffer, rec);
+            buffer_size += rec;
+        }
+        
+        if(status)
+        {
+            status = false;
+          // writing bytes to file
+            int write_bytes = write(file_fd, write_buf, buffer_size);
+            if (write_bytes == -1)
+            {
+                perror("error writing to file");
+                return -1;
+            }
+            file_size = file_size+ write_bytes;
+
+            lseek(file_fd, 0, SEEK_SET);
+
+            char *read_buf = (char *) malloc(sizeof(char) * file_size);
```

Comment:
@swapnilghonge what assumption are you making about the total size of all packets here?